cmd/ci-secret-generator/main_test.go

```diff
@@ -0,0 +1,391 @@
+						{
+							Name: "Attachment1dev",
+							Cmd:  "echo -n Attachment1dev",
+						},
+					},
+					Attribute: fieldGenerator{
+						Name: "password1dev",
+						Cmd:  "echo -n password1dev",
+					},
+					Notes: "Note1dev",
+					Params: map[string][]string{
+						"FieldNum": {
+							"1",
+							"2",
+						},
+						"Env": {
+							"dev",
+							"prod",
+						},
+					},
+				},
+				{
+					ItemName: "Item1prod",
+					Fields: []fieldGenerator{
+						{
+							Name: "Field1prod",
+							Cmd:  "echo -n Field1prod",
+						},
+					},
+					Attachments: []fieldGenerator{
+						{
+							Name: "Attachment1prod",
+							Cmd:  "echo -n Attachment1prod",
+						},
+					},
+					Attribute: fieldGenerator{
+						Name: "password1prod",
+						Cmd:  "echo -n password1prod",
+					},
+					Notes: "Note1prod",
+					Params: map[string][]string{
+						"FieldNum": {
+							"1",
+							"2",
+						},
+						"Env": {
+							"dev",
+							"prod",
+						},
+					},
+				},
+				{
+					ItemName: "Item2dev",
+					Fields: []fieldGenerator{
+						{
+							Name: "Field2dev",
+							Cmd:  "echo -n Field2dev",
+						},
+					},
+					Attachments: []fieldGenerator{
+						{
+							Name: "Attachment2dev",
+							Cmd:  "echo -n Attachment2dev",
+						},
+					},
+					Attribute: fieldGenerator{
+						Name: "password2dev",
+						Cmd:  "echo -n password2dev",
+					},
+					Notes: "Note2dev",
+					Params: map[string][]string{
+						"FieldNum": {
+							"1",
+							"2",
+						},
+						"Env": {
+							"dev",
+							"prod",
+						},
+					},
+				},
+
+				{
+					ItemName: "Item2prod",
+					Fields: []fieldGenerator{
+						{
+							Name: "Field2prod",
+							Cmd:  "echo -n Field2prod",
+						},
+					},
+					Attachments: []fieldGenerator{
+						{
+							Name: "Attachment2prod",
+							Cmd:  "echo -n Attachment2prod",
+						},
+					},
+					Attribute: fieldGenerator{
+						Name: "password2prod",
+						Cmd:  "echo -n password2prod",
+					},
+					Notes: "Note2prod",
+					Params: map[string][]string{
+						"FieldNum": {
+							"1",
+							"2",
+						},
+						"Env": {
+							"dev",
+							"prod",
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			outputBwItems, err := processBwParameters(tc.inputBwItems)
+			sort.Slice(outputBwItems, func(p, q int) bool {
+				return outputBwItems[p].ItemName < outputBwItems[q].ItemName
+			})
+			equalError(t, tc.expectedError, err)
+			equal(t, tc.expected, outputBwItems)
+		})
+	}
+}
+
+func equalError(t *testing.T, expected, actual error) {
+	if expected != nil && actual == nil || expected == nil && actual != nil {
+		t.Errorf("expecting error %v, got %v", expected, actual)
+	}
+	if expected != nil && actual != nil && expected.Error() != actual.Error() {
+		t.Errorf("expecting error msg %q, got %q", expected.Error(), actual.Error())
+	}
+}
+
+func equal(t *testing.T, expected, actual interface{}) {
```

Comment:
What utility does this serve?

Reply:
line 333 - to compare the output of the function with the expected output

Reply:
Right, but why do you feel like you need this method? Just use the actual code underneath.

Reply:
right now I would be using it in only 2 places, but as we add more tests I would keep writing the extra "t.Errorf("actual differs from expected:\n%s", cmp.Diff(expected, actual))" instead of just calling the function.